### ISSUE TITLE
Add options to choose if a tap on media files should preview inside OC or open with external apps

### ISSUE THREAD

**Issue by ghost** [owncloud/android#775]
Currently when clicking on a media file (movie and music) the integrated media player is opened.
As Android user I want to use my favorite players by default (except photo because  it displays (or is planned to display) smaller versions of images).

This is somewhat similar to: https://github.com/owncloud/android/issues/376

**Comment by xylo** [comment]
I absolutely agree with LukeOwncloud! Allowing external players to play video and audio streams would solve a lot of problems:
* much more video and audio formats supported
* much better navigation, volume, and brightness control
* full screen mode
* direct streaming of videos instead of downloading and playing

Moreover I would suggest to give the user the choice of downloading the video/audio file or streaming it.  The internal player could be actually removed to simplify the source code.

**Comment by AndyScherzinger** [comment]
This should be an easy one I case it is decided to implement this. It should be simply calling an intent, e.g. like it is described here: http://stackoverflow.com/questions/15807729/android-play-video-through-intent
This might be working for streaming to but that would need some further investigation.

@jancborchardt @rperezb what is the general strategy for media playing in ownCloud?
I see benefits for both solutions (integrated Vs. standalone) but would personally prefer the proposal of this ticket since it imho follows Androids way of handling such use cases (as in firing an intent and let other apps handle it due to them being focus on the use case, e.g. video playback).

**Comment by tobiasKaminsky** [comment]
@AndyScherzinger The "intent sending" is already there, see "open with".
So it would only be one Preference and if it is set, then the normal click would show the same as "open with".

**Comment by AndyScherzinger** [comment]
Ah, mist that one...  True!

**Comment by xylo** [comment]
@tobiasKaminsky Thanks for the hint to "open with". What's what I was looking for.

**Comment by jancborchardt** [comment]
Yeah, I would handle media playing (music as well as videos) via intents, offloading it to other apps. Or we do it very very good. Currently that’s not the case at least for music, so there we should definitely offload.

cc @rperezb @davivel

**Comment by davivel** [comment]
Let me rephrase the title.

**Comment by DeepDiver1975** [comment]
🤖 Automated issue triage by Claude Code review agent.

This issue was opened before 2020 and has had no activity since then, so we are
closing it as outdated. If this is still relevant on a current ownCloud release,
please **reopen** (or open a fresh issue) with up-to-date details — thank you\!
